test/e2e/framework/statefulset_utils.go

```diff
@@ -812,6 +821,65 @@ func NewStatefulSet(name, ns, governingSvcName string, replicas int32, statefulP
 	}
 }
 
+// NewStatefulSetV1beta2 creates a new NGINX v1beta2 StatefulSet for testing. The StatefulSet is named name, is in namespace ns,
+// statefulPodsMounts are the mounts that will be backed by PVs. podsMounts are the mounts that are mounted directly
+// to the Pod. labels are the labels that will be usd for the StatefulSet selector.
+func NewStatefulSetV1beta2(name, ns, governingSvcName string, replicas int32, statefulPodMounts []v1.VolumeMount, podMounts []v1.VolumeMount, labels map[string]string) *appsV1beta2.StatefulSet {
```

Comment:
I don't think you need to add code to handle v1beta2 at this point. You can create a v1beta1 StatefulSet as before, and it should still work to interact with its /scale subresource via the v1beta2 API. The server side does conversions to serve the same underlying set of objects under whatever version of the API the client requests.